lib/model/realm.dart

```diff
@@ -432,6 +460,26 @@ class RealmStoreImpl extends HasUserGroupStore with RealmStore {
     return displayFields.followedBy(nonDisplayFields).toList();
   }
 
+  static List<ThumbnailFormat> _filterAndSortThumbnailFormats(
+    List<ThumbnailFormat> initialServerThumbnailFormats, {
+    required bool animated,
+  }) {
+    return initialServerThumbnailFormats
+      .where((format) => format.animated == animated)
+      .toList()
+      ..sort(_compareThumbnailFormats);
+  }
+
+  /// A comparator to sort formats by max resolution, ascending.
+  ///
+  /// "Max resolution" means
+  /// [ThumbnailFormat.maxWidth] * [ThumbnailFormat.maxHeight].
+  static int _compareThumbnailFormats(ThumbnailFormat a, ThumbnailFormat b) {
+    final aMaxResolution = a.maxWidth * a.maxHeight;
+    final bMaxResolution = b.maxWidth * b.maxHeight;
+    return aMaxResolution - bMaxResolution;
```

Comment:
```suggestion
    return aMaxResolution.compareTo(bMaxResolution);
```

Otherwise this would go wrong if the values differed by half or more of the max range of the int type.

(Not likely for this particular data; but best to set an example of the pattern that is consistently correct.)